File: thurgadina/Agent of Change.pl
#agent of change by osirous

my $expedition_name = "Icewell Keep";
my $min_players     = 1;
my $max_players     = 40;
my $dz_zone         = "thurgadinb";
my $dz_version      = 0;
my $dz_duration     = 28800; # 8 hours | #3600; # 1 hour

sub EVENT_SAY {
  if ($text =~ /hail/i) {
    my $dz = $client->GetExpedition();
    if ($dz && $dz->GetName() eq $expedition_name) {
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
    else {
      quest::say("Would you like to [" . quest::saylink("request") . "] the expedition?");
    }
  }
  elsif ($text =~ /request/i) {
    my $dz = $client->CreateExpedition($dz_zone, $dz_version, $dz_duration, $expedition_name, $min_players, $max_players);
    if ($dz) {
      $dz->SetSafeReturn("thurgadina", -2.01, 157.24, 3.38, 252.25); # outside portal
      $dz->SetZoneInLocation(6.95, 162.89, 3.38, 508.25); # normal zone in place
      $dz->AddReplayLockout(32400); # immediately add a 9 hour replay lockout on creation
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
  }
  elsif ($text =~ /ready/i) {
    $client->MovePCDynamicZone("thurgadinb");
  }
}

File: kael/Agent_of_Change.lua
local aoc_raid = {
    expedition = { name="Wakening Lands", min_players=1, max_players=72},
    instance   = { zone="wakening", version=0, duration=eq.seconds("8h") }, -- zone lasts 8 hours
    safereturn = { zone="kael", x=528.11, y=-149.99, z=-196.88, h=227.0 }, --outside portal
    zonein     = { x=3167.90, y=-228.84, z=-394.21, h=260 },
}

function event_say(e)
	local dz = e.other:GetExpedition()
	if(e.message:findi("hail")) then
		if dz.valid and dz:GetZoneName() == aoc_raid.instance.zone then
			e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
		else
			e.self:Say("Would you like to [" .. eq.say_link("request") .. "] the expedition?")
		end
	elseif(e.message:findi("request")) then
		local dz = e.other:CreateExpedition(aoc_raid)
		dz:AddReplayLockout(eq.seconds("9h")) -- 9 hour lockout
		e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
	elseif(dz:GetZoneName() == aoc_raid.instance.zone and e.message:findi("ready")) then
		e.other:MovePCDynamicZone(aoc_raid.instance.zone)
	end
end